android/app/src/main/java/app/organicmaps/widget/SearchToolbarController.java

```diff
@@ -210,14 +211,22 @@ public void showSearchControls(boolean show)
     UiUtils.showIf(show, mSearchContainer);
   }
 
-  public void onActivityResult(int requestCode, int resultCode, Intent data)
+  public void onVoiceRecognitionResult(ActivityResult activityResult)
   {
-    if (requestCode == REQUEST_VOICE_RECOGNITION && resultCode == Activity.RESULT_OK)
+    UiUtils.handleActivityResult(new HandleActivityResult()
```

Comment:
We don't need to handle request codes. Because, when we launch an activity we pass what to do when a particular activity returns, as a callback with it, when we register it. So, it always execute the related logic.